### ISSUE TITLE
rustfmt deletes vertical bar for match branches!

### ISSUE THREAD

**Issue by rusterize** [rust-lang-nursery/rustfmt#2462]
The vertical bar for match brances [was just stabilized](https://github.com/rust-lang/rust/pull/47947), but `rustfmt` deletes them.

Thanks for the great work.

**Comment by topecongiro** [comment]
The stabilization PR removed `beginning_vert` field from `ast::Arm`. We need to bring it back to support it in rustfmt.

**Comment by topecongiro** [comment]
Note that rustfmt will not remove the vertical bar at the beginning of march arms, but it will not format the entire arm with the beginning vertical bar.

@rusterize Would you tell us which version of rustfmt you are using please? I think you might be using older version of rustfmt.

**Comment by rusterize** [comment]
```
$ rustfmt --version
0.3.8-nightly (346238f 2018-02-04)
```

**Comment by rusterize** [comment]
Before rustfmt:
```rust
        match anchor {
            | &Some(ref row_index) => Ok(row_nodes_borrow
                .get(row_index)
                .ok_or(RowIndexNotFound(format!("{}", row_index)))?),

            | &None => Ok(&self.root_node_iter),
        }
```
after rustfmt:
```rust
        match anchor {
            &Some(ref row_index) => Ok(row_nodes_borrow
                .get(row_index)
                .ok_or(RowIndexNotFound(format!("{}", row_index)))?),
            |
            &None => Ok(&self.root_node_iter),
        }
```

**Comment by topecongiro** [comment]
@rusterize Thank you for sharing the code! I could confirm the issue. And it should be fixed with #2464.

**Comment by nrc** [comment]
I'm not 100% sure what Rustfmt should be doing here, see https://github.com/rust-lang-nursery/fmt-rfcs/issues/119, however, it is clear the current behaviour is bad so I've merged the fixing PR.
